build-system/pr-check.js

```diff
@@ -286,16 +286,17 @@ function runAllCommands() {
     command.runVisualDiffTests();
     command.runJsonAndLintChecks();
     command.runDepAndTypeChecks();
-    command.runUnitTests();
-    // command.testDocumentLinks() is skipped during push builds.
     command.buildValidatorWebUI();
     command.buildValidator();
+    command.runUnitTests(); // Requires validator to be built.
+    // command.testDocumentLinks() is skipped during push builds.
   }
   if (process.env.BUILD_SHARD == "integration_tests") {
     command.cleanBuild();
     command.buildRuntimeMinified();
     command.runPresubmitTests();  // Needs runtime to be built and served.
-    command.runIntegrationTests();
+    command.buildValidator();
```

Comment:
Actually, looks like the test doesn't run on sauce labs anyways (`skipSauceLabs()`), so we can move it out of the `test/integration` folder and run it as a unit test instead. Doing that now.